src/Datadog.Trace/DuckTyping/DuckTypeExtensions.cs

```diff
@@ -22,7 +27,109 @@ public static T As<T>(this object instance)
         /// <param name="instance">Object instance</param>
         /// <param name="targetType">Target type</param>
         /// <returns>DuckType instance</returns>
-        public static object As(this object instance, Type targetType)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static object DuckCast(this object instance, Type targetType)
             => DuckType.Create(targetType, instance);
+
+        /// <summary>
+        /// Tries to ducktype the object implementing a base class or interface T
+        /// </summary>
+        /// <typeparam name="T">Target type</typeparam>
+        /// <param name="instance">Object instance</param>
+        /// <param name="value">Ducktype instance</param>
+        /// <returns>true if the object instance was ducktyped; otherwise, false.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryDuckCast<T>(this object instance, out T value)
+        {
+            try
+            {
+                value = DuckType.Create<T>(instance);
+                return true;
```

Comment:
Should we call `DuckType.CanCreate<T>(instance)` first to avoid throwing if we know it won't work? I'm not sure what the performance implications are for the common case (when we can create). Even if we do this, we probably want to keep the try/catch because `TryFoo()` methods should never throw.
```suggestion
                if(DuckType.CanCreate<T>(instance))
                {
                    value = DuckType.Create<T>(instance);
                    return true;
                }

                value = default;
                return false;
```

Reply:
I'd prefer to avoid calling `CanCreate` for performance reasons, that's why in this case is like a best effort approach. Also as you said, we must keep the try/catch so I don't think it worth it.

Reply:
I see, thanks for clarifying!